core/src/system_monitor_service.rs

```diff
@@ -223,12 +252,61 @@ pub fn verify_net_stats_access() -> Result<(), String> {
     Ok(())
 }
 
+#[cfg(target_os = "linux")]
+fn read_disk_stats() -> Result<DiskStats, String> {
+    let file_path_diskstats = PROC_DISKSTATS_PATH;
+    let file_diskstats = File::open(file_path_diskstats).map_err(|e| e.to_string())?;
+    let mut reader_diskstats = BufReader::new(file_diskstats);
+    parse_disk_stats(&mut reader_diskstats)
+}
+
+#[cfg_attr(not(target_os = "linux"), allow(dead_code))]
+fn parse_disk_stats(reader_diskstats: &mut impl BufRead) -> Result<DiskStats, String> {
+    let mut stats = DiskStats::default();
+    let mut num_disks = 0;
+    for line in reader_diskstats.lines() {
+        let line = line.map_err(|e| e.to_string())?;
+        let values: Vec<_> = line.split_ascii_whitespace().collect();
+
+        if values.len() != 20 {
+            return Err("parse error, expected exactly 20 disk stat elements".to_string());
+        }
+        if values[2].starts_with("loop") || values[1].ne("0") {
+            // Filter out the loopback io devices.
+            // Only look at raw device (filter partitions)
+            continue;
+        }
+
+        num_disks += 1;
+        stats.reads_completed += values[3].parse::<u64>().map_err(|e| e.to_string())?;
```

Comment:
similarly, should we be totally bailing or just `continue`ing

Reply:
I'm thinking it would be better to not get any metrics rather than potentially report incorrect metrics